### PR TITLE
Fixing EventStoreCatchUpSubscription.php

### PR DESCRIPTION
**Pull request by burzum**

### CODE REVIEW

File: src/Internal/EventStoreCatchUpSubscription.php [hunk 1]
```diff
@@ -163,37 +162,19 @@ private function loadHistoricalEvents(): void
...
-                }
-
-                public function __invoke(
-                    EventStoreSubscription $subscription,
-                    SubscriptionDropReason $reason,
-                    ?Throwable $exception = null
-                ): void {
-                    ($this->callback)($reason, $exception);
-                }
+            $subscriptionDropped = function(
```
Comment (enumag) [inline src/Internal/EventStoreCatchUpSubscription.php:172]:
```suggestion
            $subscriptionDropped = function (
```

File: src/Internal/EventStoreCatchUpSubscription.php [hunk 2]
```diff
@@ -163,37 +162,19 @@ private function loadHistoricalEvents(): void
...
-                    $this->callback = $callback;
-                }
-
-                public function __invoke(
-                    EventStoreSubscription $subscription,
-                    ResolvedEvent $resolvedEvent
-                ): void {
-                    ($this->callback)($subscription, $resolvedEvent);
-                }
+            $eventAppeared = function(
```
Comment (enumag) [inline src/Internal/EventStoreCatchUpSubscription.php:165]:
```suggestion
            $eventAppeared = function (
```

File: src/Internal/EventStoreCatchUpSubscription.php [hunk 3]
```diff
@@ -55,9 +53,9 @@ abstract class EventStoreCatchUpSubscription implements SyncEventStoreCatchUpSub
     private ?DropData $dropData;
     private bool $allowProcessing;
     private bool $isProcessing;
-    protected bool $shouldStop;
-    private bool $isDropped;
-    private bool $stopped;
+    protected bool $shouldStop = false;
+    private bool $isDropped = false;
+    private bool $stopped = false;
```
Comment (enumag) [inline src/Internal/EventStoreCatchUpSubscription.php]: This property is already initialized in the constructor and also it's initialized with true there. So this doesn't make sense.

**Review by enumag — CHANGES_REQUESTED** [review]

File: src/Internal/EventStoreCatchUpSubscription.php (hunk 3 above)
Comment (prolic) [reply src/Internal/EventStoreCatchUpSubscription.php]: Indeed, the initial state of the subscription is stopped, not started.

File: src/Internal/EventStoreCatchUpSubscription.php [hunk 4]
```diff
@@ -163,37 +162,19 @@ private function loadHistoricalEvents(): void
...
-                    EventStoreSubscription $subscription,
-                    ResolvedEvent $resolvedEvent
-                ): void {
-                    ($this->callback)($subscription, $resolvedEvent);
-                }
+            $eventAppeared = function (
+                EventStoreSubscription $subscription,
+                ResolvedEvent $resolvedEvent
+            ): void {
+                ($this->callback)($subscription, $resolvedEvent);
```
Comment (prolic) [inline src/Internal/EventStoreCatchUpSubscription.php:169]: What is `$this->callback` ? No such property exists.

File: src/Internal/EventStoreCatchUpSubscription.php [hunk 5]
```diff
@@ -163,37 +162,19 @@ private function loadHistoricalEvents(): void
...
-                    ?Throwable $exception = null
-                ): void {
-                    ($this->callback)($reason, $exception);
-                }
+            $subscriptionDropped = function (
+                EventStoreSubscription $subscription,
+                SubscriptionDropReason $reason,
+                ?Throwable $exception = null
+            ): void {
+                ($this->callback)($reason, $exception);
```
Comment (prolic) [inline src/Internal/EventStoreCatchUpSubscription.php:177]: Same here, what is `$this->callback` ?

**Review by prolic — CHANGES_REQUESTED** [review]

File: src/Internal/EventStoreCatchUpSubscription.php (hunk 4 above)
Comment (burzum) [reply src/Internal/EventStoreCatchUpSubscription.php:169]: You tell me. It was there before. :) Not sure from where it came from and how to substitute it now. My example app still works and doesn't complain about anything for now.

File: src/Internal/EventStoreCatchUpSubscription.php (hunk 5 above)
Comment (burzum) [reply src/Internal/EventStoreCatchUpSubscription.php:177]: You tell me. It was there before. :) Not sure from where it came from and how to substitute it now. My example app still works and doesn't complain about anything for now.

File: src/Internal/EventStoreCatchUpSubscription.php (hunk 3 above)
Comment (burzum) [reply src/Internal/EventStoreCatchUpSubscription.php]: When I was running the code php complained about this line as well. I've changed the value and going to push it.

File: src/Internal/EventStoreCatchUpSubscription.php (hunk 4 above)
Comment (enumag) [reply src/Internal/EventStoreCatchUpSubscription.php:169]: Check the argument that was previously passed to the anonymous class constructor. That's what you should use here.

Comment (burzum) [reply src/Internal/EventStoreCatchUpSubscription.php:169]: Guess I need to add the callback as first arg of the anonymous function? I'm not sure how the callback is now passed.

https://github.com/prooph/event-store-http-client/blob/master/src/Internal/EventStoreCatchUpSubscription.php#L166-L180

**Comment by prolic** [comment]
see here: https://github.com/prooph/event-store-http-client/pull/54

I was in a hurry, maybe I missed something, but it should look like this, more or less

**Comment by prolic** [comment]
Feel free to add commits to my branch, if I missed something

**Pull request closed without merging**